### PR TITLE
Feat/1414 fac filters page

### PR DESCRIPTION
**Pull request by denisonishchenkou**
# Description

- Loading Overlay component
- FAC filters page example (work in progress)

Fixes # (issue number)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Make sure to do not repeat yourself (DRY)
- [x] Snapshots tested
- [ ] A11y tested
- [x] CSS utilises BEM naming convention and structure
- [x] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] Crossbrowser testing (IE11, Safari 8+, iOS 8.4+, Android 4.4+, Firefox ESR (v52.x), iPhone (4s,6), iPad 2, Galaxy s5) 
- [x] My changes generate no new warnings
- [x] I have added tests (e2e and unit) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Added the new component to the index.js (Vue and Lib) export

# Reviewer Checklist

- [ ] Code is not repeated (DRY)
- [ ] ES6+ code is used where possible
- [ ] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] CSS utilises BEM naming convention and structure
- [ ] Snapshots tested against a copy of dev snapshots to check UI changes are intended
- [ ] A11y tested
- [ ] Crossbrowser tested in at least IE11

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit ecda693450ca21347e7ab6cb9e80697c0e77b979

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/5e820b157a9f980009e4a7f0

**Comment by denisonishchenkou** [comment]
Hi @gepifanio could you please review `LoadingOverlay` component.
And also I wanted to ask about styles of the filters box (in the left sidebar), I've added styles to the page example component. Maybe it's better to create Vue component for this?

File: components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            :key="resultIndex">
+            <GenericCard
+              :cols="2"
+              :title="result.name" />
+          </ListItem>
+        </div>
+      </LoadingOverlay>
+    </div>
+    <div slot="side">
+      <div class="bg-light-blue filter-block">
```
Comment (gepifanio) [inline components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue:24]: Your question about this:

Yes create a component called `FilterBox` or `FilterDropdown`

Place inside of the `filters/` folder structure, I also think all the other filters related components (results, dropdown..) should also be inside of that folder structure, and add the styles in there.

But make sure to the component be full `width`, so the page control the width of the component and accept the `Filter by` and `course types` text via prop, so it is reusable cross any other system.

**Comment by denisonishchenkou** [comment]
@gepifanio Could you please help me with questions:

Update filters button - how should it work? In this: https://handbook.unimelb.edu.au/search example after applying any filter the button doesn't do anything.

I've added update handler on the "Apply filter" button inside the dropdown and If I understood everything correctly then I think we don't need the "update filters" button.

Clear filters - should this button reset filters to defaults or should it clear all filters?

if it clears all filters than what data should be shown on a page when we don't have selected checkboxes?

We have "All" checkbox and it looks like we should have it checked by default. Or maybe we should block the "Apply filters" button when nothing is selected.

**Comment by denisonishchenkou** [comment]
@gepifanio I've grouped Loading and Filter components and updated readme files. Could you please take a look?

Also I wanted to ask about the segmentation notice (above the result tiles):
As far as I understood from https://app.zenhub.com/workspaces/fac-rebuild-5ce4e45c3c14bf6d63636aa1/issues/unimelb/find-a-course/1689 when custom filters applied, the user should have the possibility to reset them. When default filters applied, the modal from find a course project should be opened on click.
Do I need to implement that modal in the pattern-lib?

**Comment by jonathan-doc** [comment]
> Update filters button - how should it work? In this: https://handbook.unimelb.edu.au/search example after applying any filter the button doesn't do anything.
> 
> I've added update handler on the "Apply filter" button inside the dropdown and If I understood everything correctly then I think we don't need the "update filters" button.

Results shouldn't update every time a filter facet is checked (as that can get quite annoying). When the user makes a selection inside a dropdown filter, it shouldn't update the results unless:
i. the user hits 'apply filter' or 
ii. if they click out of / close the dropdown (as discussed with @jaredh-um, so that there won't be items checked in the dropdown that aren't applied, and user selections aren't lost)

The 'update results' button should update the filter results based on the filters. The button might not get used much because of the behaviour described above (i.e. clicking out of a dropdown filter triggering a change), but if this does change, the button will be necessary.

> Clear filters - should this button reset filters to defaults or should it clear all filters?

The clear filters button should just clear the filters (e.g. reset them to 'All'); defaults should be handled by segmentation.

> if it clears all filters than what data should be shown on a page when we don't have selected checkboxes?
>
> We have "All" checkbox and it looks like we should have it checked by default. Or maybe we should block the "Apply filters" button when nothing is selected.

If no checkboxes are selected, the 'All' checkbox should be selected.

**Comment by jonathan-doc** [comment]
> Also I wanted to ask about the segmentation notice (above the result tiles):
> As far as I understood from https://app.zenhub.com/workspaces/fac-rebuild-5ce4e45c3c14bf6d63636aa1/issues/unimelb/find-a-course/1689 when custom filters applied, the user should have the possibility to reset them. When default filters applied, the modal from find a course project should be opened on click.
> Do I need to implement that modal in the pattern-lib?

@gepifanio correct me if I'm wrong–

The segmentation notice is specific to FAC and is only appended to the results summary in FAC. 

Results summary (e.g.): "7 results found with 1 filter applied"
Segmentation notice (e.g.): "(based on Domestic undergraduate - change)."

**Comment by gepifanio** [comment]
> > Also I wanted to ask about the segmentation notice (above the result tiles):
> > As far as I understood from https://app.zenhub.com/workspaces/fac-rebuild-5ce4e45c3c14bf6d63636aa1/issues/unimelb/find-a-course/1689 when custom filters applied, the user should have the possibility to reset them. When default filters applied, the modal from find a course project should be opened on click.
> > Do I need to implement that modal in the pattern-lib?
> 
> @gepifanio correct me if I'm wrong–
> 
> The segmentation notice is specific to FAC and is only appended to the results summary in FAC.
> 
> Results summary (e.g.): "7 results found with 1 filter applied"
> Segmentation notice (e.g.): "(based on Domestic undergraduate - change)."

Thats right

**Comment by gepifanio** [comment]
> @gepifanio I've grouped Loading and Filter components and updated readme files. Could you please take a look?
> 
> Also I wanted to ask about the segmentation notice (above the result tiles):
> As far as I understood from https://app.zenhub.com/workspaces/fac-rebuild-5ce4e45c3c14bf6d63636aa1/issues/unimelb/find-a-course/1689 when custom filters applied, the user should have the possibility to reset them. When default filters applied, the modal from find a course project should be opened on click.
> Do I need to implement that modal in the pattern-lib?

- Filtered results component -> should be inside the filters folder
- Segmentation notice will be implemented in the Study FAC project
- For the readme turns the source and/or HTML to true, so the CMS users will be able to copy the markup of the components and use it in any other page. 

So to clarify:

the `pattern-lib` it's where sits dumb components, and those components are used across different systems and platforms such as: 

- CMS: They copy the raw HTML markup to use those components
- Vue App: which use the component and its props

So everything that goes in `pattern-lib`, should be created to be generic. So those filter designs that you have, are the ones that will be used in the `Study FAC` project, however this filters need to be generic for any other use case as well. 

The question about the `modal` you don't need to recreate it, and the segmentation message and the filter defaults, don't need to be implemented in `pattern-lib`, you will be implementing those in the `Study FAC` using Vue components imported from the `pattern-lib`. 

So in summary: 
`pattern-lib` dumb generic components.
`Study FAC` where will use those components to create the design page you have received.

If you wanna set up a call, it might be better to explain, how everything will works.

**Comment by denisonishchenkou** [comment]
@jonathan-doc 
> Results shouldn't update every time a filter facet is checked (as that can get quite annoying). When the user makes a selection inside a dropdown filter, it shouldn't update the results unless:
i. the user hits 'apply filter' or
ii. if they click out of / close the dropdown (as discussed with @jaredh-um, so that there won't be items checked in the dropdown that aren't applied, and user selections aren't lost)

Is it OK that the user doesn't have the possibility to cancel his last changes? I thought that default behavior for this kind of elements is to cancel recent changes when the user clicks outside the dropdown.
But I still did as you asked, you can check in this branch.

> If no checkboxes are selected, the 'All' checkbox should be selected.

Can we change this behavior to just disabling the "Apply filters" button? If the "All" checkbox is already selected and the user clicks it second time - nothing happens, as for me, this is a strange behavior

@gepifanio I've moved FilteredResults component to the filters folder, removed segmentation component and updated readme files. Could you please take a look?

**Comment by gepifanio** [comment]
> @gepifanio I've moved FilteredResults component to the filters folder, removed segmentation component and updated readme files. Could you please take a look?

- Looks good, its just missing how to use the component `Filtered Results` as a Vue component in the `readme` docs. 


Regarding the example page:

- Please setup the mock data using real simulation data instead of random result such as:

mock data:

```
  {
    id: '1',
    name: 'Anthropology',
    cols: 2,
    excerpt: "Bachelor",
  },
```

wheres the `excerpt` will be one of the options to be filtered as an example, so we can simulate better how it works, instead of using random results for that. 

in this case, excerpt could be: `Concurrent Diploma`, `Master`, `Graduate Certificate`, `Doctoral`... any of that options, and when we click in the filter should filter the correct items. 

- The numbers that show in the filter `label` should be a count of the options under that "type" instead of hardcoded in the label content.

File: components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            :items="results.length"
+            :filters="filtersApplied"
+            :secondary-message="secondaryMessage"
+            :callback="segmentationChange">
+            <div class="grid grid--center grid--2col">
+              <ListItem
+                v-for="item in results"
+                :key="item.id">
+                <GenericCard
+                  :thumb="item.thumb"
```
Comment (gepifanio) [inline components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue:27]: as you are not using some of the props you can remove from the code:

`thumb`, `href` and `tags`

File: components/filters/filter-box/FilterBox.vue [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        class="btn--cta"
+        @click.native.prevent="onUpdateResults">
+        Update results
+      </ButtonIcon>
+    </div>
+  </div>
+</template>
+
+<script>
+import ButtonIcon from '../../buttons/ButtonIcon.vue';
```
Comment (gepifanio) [inline components/filters/filter-box/FilterBox.vue:34]: you can change the import to just use `'components/buttons/ButtonIcon.vue'` instead

File: components/filters/filter-box/stories/FilterBoxAside.vue [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        :options="options"
+        filter-by="Course types"
+        placeholder-label="course types"
+        options-label="Course types to include:" />
+    </div>
+  </SectionTwoCol>
+</template>
+
+<script>
+import SectionTwoCol from '../../../section/SectionTwoCol.vue';
```
Comment (gepifanio) [inline components/filters/filter-box/stories/FilterBoxAside.vue:19]: you can change the import to just use `'components/section/SectionTwoCol.vue'` instead

File: components/filters/filter-dropdown/FilterDropdown.vue [hunk 5]
```diff
@@ -63,8 +63,9 @@
 import cloneDeep from 'lodash.clonedeep';
 import NestedCheckbox from './components/NestedCheckbox/index.vue';
```
Comment (gepifanio) [inline components/filters/filter-dropdown/FilterDropdown.vue:64]: you can change the import to just use `'components/NestedCheckbox/index.vue'` instead

File: components/filters/filter-dropdown/FilterDropdown.vue [hunk 6]
```diff
@@ -63,8 +63,9 @@
...
 import optionsValidator from './nestedCheckboxOptionsValidator';
-import SvgIcon from '../icons/SvgIcon.vue';
-import ButtonIcon from '../buttons/ButtonIcon.vue';
+import SvgIcon from '../../icons/SvgIcon.vue';
```
Comment (gepifanio) [inline components/filters/filter-dropdown/FilterDropdown.vue:66]: you can change the import to just use `'components/icons/SvgIcon.vue'` instead

File: components/filters/filter-dropdown/FilterDropdown.vue [hunk 7]
```diff
@@ -63,8 +63,9 @@
...
+import ButtonIcon from '../../buttons/ButtonIcon.vue';
```
Comment (gepifanio) [inline components/filters/filter-dropdown/FilterDropdown.vue:67]: you can change the import to just use `'components/buttons/ButtonIcon.vue'` instead

File: components/loader/_loader-overlay.css [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    bottom: 0;
+    left: 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition-property: opacity, transform;
+    transition-duration: var(--transition-speed-normal);
+    opacity: 0;
+    transform: translateY(-30px);
+    z-index: 1;
```
Comment (gepifanio) [inline components/loader/_loader-overlay.css:19]: Use the variable for z-index inside of the `_vars.css` file 

`z-index: var(--z-default);`

File: components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+Props:
+```
+isLoading: true, // boolean, required
+spinnerText: 'Fetching results', // string
+isSpinnerVisible: false, // boolean, true by default
+```
+
+Overlay without spinner:
+```vue
```
Comment (gepifanio) [inline components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md:9]: - quicky tip: most of the times in the story if you use the `html` = true it already show how to use the `Vue` component so you don't need to write this piece of code, but sometimes it hides the objects, and need to be written down like that. 

```
export default {
  readme: { custom: "", source: , html: true },
};
```

In this case you can just delete this `vue` section and turn the `html` to true it will show the same ;)

File: components/filters/filter-dropdown/FilterDropdown.vue (hunk 5 above)
Comment (denisonishchenkou) [reply components/filters/filter-dropdown/FilterDropdown.vue:64]: `NestedCheckbox` is used only in the `FilterDropdown` component and I placed it inside the `components/filters/filter-dropdown/` folder and didn't make it globally accessible. Do you think it should be global component?

File: components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md (hunk 9 above)
Comment (denisonishchenkou) [reply components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md:9]: I didn't use `html` options because in some stories I used additional components and buttons to show how to interact with the component or to remove duplicates and I wrote custom examples to remove those helpers from readme.

File: components/loader/_loader-overlay.css (hunk 8 above)
Comment (denisonishchenkou) [reply components/loader/_loader-overlay.css:19]: fixed

File: components/filters/filter-dropdown/FilterDropdown.vue (hunk 7 above)
Comment (denisonishchenkou) [reply components/filters/filter-dropdown/FilterDropdown.vue:67]: fixed

File: components/filters/filter-dropdown/FilterDropdown.vue (hunk 6 above)
Comment (denisonishchenkou) [reply components/filters/filter-dropdown/FilterDropdown.vue:66]: fixed

File: components/filters/filter-box/stories/FilterBoxAside.vue (hunk 4 above)
Comment (denisonishchenkou) [reply components/filters/filter-box/stories/FilterBoxAside.vue:19]: fixed

File: components/filters/filter-box/FilterBox.vue (hunk 3 above)
Comment (denisonishchenkou) [reply components/filters/filter-box/FilterBox.vue:34]: fixed

File: components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue (hunk 2 above)
Comment (denisonishchenkou) [reply components/pages/stories/find-a-course-interest-pages/MajorsAndSpecialisations.vue:27]: fixed

File: components/filters/filter-dropdown/FilterDropdown.vue (hunk 5 above)
Comment (gepifanio) [reply components/filters/filter-dropdown/FilterDropdown.vue:66]: It is fine just keep it there, I thought that was coming from the global source.

File: components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md (hunk 9 above)
Comment (gepifanio) [reply components/loader/stories/loading-overlay/loadingOverlayWithoutSpinner.md]: That's cool, I got what you mean, in this case, is better to not use it ;)

File: components/filters/filter-dropdown/FilterDropdown.vue (hunk 5 above)
Comment (denisonishchenkou) [reply components/filters/filter-dropdown/FilterDropdown.vue:66]: Will keep it, thanks.

**Comment by gepifanio** [comment]
> > @gepifanio I've moved FilteredResults component to the filters folder, removed segmentation component and updated readme files. Could you please take a look?
> 
> * Looks good, its just missing how to use the component `Filtered Results` as a Vue component in the `readme` docs.
> 
> Regarding the example page:
> 
> * [ ]  Please setup the mock data using real simulation data instead of random result such as:
> 
> mock data:
> 
> ```
>   {
>     id: '1',
>     name: 'Anthropology',
>     cols: 2,
>     excerpt: "Bachelor",
>   },
> ```
> 
> * [ ]  wheres the `excerpt` will be one of the options to be filtered as an example, so we can simulate better how it works, instead of using random results for that.
> * [ ]  in this case, excerpt could be: `Concurrent Diploma`, `Master`, `Graduate Certificate`, `Doctoral`... any of that options, and when we click in the filter should filter the correct items.
> * [ ]  The numbers that show in the filter `label` should be a count of the options under that "type" instead of hardcoded in the label content.
> * [ ]  Add one case in the filters which would show the `error box` as well, so we test this component
> * [ ]  In all components imports please use the `alias` to import the component, so remove the `../../../` and just add `components/...`

Hey @denisonishchenkou  please check the boxes and tag me in the PR once is all done.

**Comment by denisonishchenkou** [comment]
Hey @gepifanio I've finished all requested changes. Could you please review?

**Comment by gepifanio** [comment]
- [ ] the button's text if doesn't fit in the container should go to the next line
image


- [ ] the child options of the courses shouldn't be showing, instead of Bachelor (6) in this case show be only 1 course type selected
image


- [ ] the filter box spacing should be following the Figma specification like this screen shot
image



Inside of the filter:
- [ ] Spacing between the filter facets should be 12px
- [ ] checkbox shrinks if the text is large
- [ ] hover on each filter option should underline
image

**Comment by gepifanio** [comment]
@denisonishchenkou Some more changes:

- [ ] Change the folder structure for the filters based in:

Filters - > Components - > Filter box
Filters - > Components - > Filter dropdown
Filters - > Components - > Filtered Results
Filters - > Usage Example -> 1 Dropdown option -> Majors and specialisations tab

- [ ] Add new Usage example in the folder:
Filters - > Usage Example -> 2 Dropdown option -> Default


``` data example

{
    {
    id: '1',
    name: 'History and Philosophy of Science',
    city: [{
      name: Sydney,
      campus: ['cbd', 'bondi'],
    },
    {
      name: Melbourne,
      campus: ['southbank', 'parkville']
    }]
    faculty: 'Faculty of History',
    description: 'Labore est anim consequat veniam duis nulla esse esse labore repr'
  },
  {
    id: '2',
    name: 'Screen and Cultural Studies',
    city: [{
      name: Gold coast,
      campus: ['cbd'],
    },
    {
      name: Melbourne,
      campus: ['southbank']
    }]
    faculty: 'Faculty of Art',
    description: 'Labore est anim consequat veniam duis nulla esse esse labore repr'
  },
  {
    id: '3',
    name: 'Sociology',
    city: [{
      name: Sydney,
      campus: ['cbd', 'mainly'],
    }],
    faculty: 'Faculty of Sociology',
    description: 'Labore est anim consequat veniam duis nulla esse esse labore repr'
  },
}

```
- the first dropdown will be `Locations`
the locations could be cities and or campus

- the second dropdown will be `Faculties`
the faculties doesn't have nested children like the locations so will have only the faculty name to select. eg:
image
 
This new story example is to see how the filters are reusable in any data case/ structure.

**Comment by denisonishchenkou** [comment]
@gepifanio I have finished the changes you requested. Could you leave a review?

Could you please also review `components/filters/components/filter-dropdown/nestedCheckboxOptionsValidator.js:4`

I found useful to add console.error to the validator functions, this helps a lot during the debug process, but I'm not sure if it works when this component is used in the project (when `patern-lib` is built and used as a dependency)

**Comment by gepifanio** [comment]
@denisonishchenkou Good job in adding the validator, I am also not really sure how to add that to the bundle, probably needs to check the `webpack` + `targets/lib/index.js` / `targets/vue/index.js` I think if import then there should work.

**Comment by jonathan-doc** [comment]
- [ ]  When 'All' is selected (i.e. filter is cleared), dropdown text font-weight should be 400, and text should display 'All [filter name]' as in 'Delivery modes' example here:
image

- [ ] If filters are cleared, results summary should read '3 results found with 0 filters applied.', and dropdowns should display as described above. E.g.:
image

- [ ] Each facet should indicate their total result count. So in this example, 'All' should read 'All (3)', and each of the Faculties should be followed by (1) – e.g. 'Faculty of History (1)':
image

- [ ] Results summary should indicate how many filters are applied. Currently, they show how many facets are selected in the first filter; so if you select 4 Locations and any number of Faculties, it always displays '3 results found with 4 filters applied.'. There are only two filters, so it should actually say '3 results with 2 filters applied' in this case.

- [ ] Dropdown 'Clear filter' buttons are clearing _all_ filters; they should only clear their own filter.

- [ ] Spacing between dropdowns should be 16px / 1rem.

- [ ] Review spacing between facets and dropdown buttons (the padding-top (1.5rem) is correct, but there is an extra gap between them):
image
 vs 
image

- [ ] If a dropdown menu is open, opening another dropdown should close the other.

File: components/filters/stories/Default/Default.vue [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+          <FilteredResults
+            :items="results.length"
+            :filters="filtersApplied">
+            <div class="grid grid--center grid--2col">
+              <ListItem
+                v-for="item in results"
+                :key="item.id">
+                <GenericCard
+                  :title="item.name"
+                  :excerpt="item.excerpt"
```
[the remark this answers is not in the document]
Comment (denisonishchenkou) [reply components/filters/stories/Default/Default.vue:26]: changed

File: components/filters/stories/Default/Default.vue [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+                  :cols="2" />
```
[the remark this answers is not in the document]
Comment (denisonishchenkou) [reply components/filters/stories/Default/Default.vue:27]: @gepifanio I have a question about the mock results: we have 3 options, but each option has multiple locations (city/campus pairs). I implemented filtering in such a way that the maximum result that the user can see is 3 tiles (as in the mock array). But in this case, I do not know what location I should render in the tile. Maybe each mock data item needs to be duplicated for each location? So, for example, when a user selects Sydney/cbd and Sydney/bondi, he will see two “Faculty of History” tiles instead of one. Does this make sense?

[the remark this answers is not in the document]
Comment (gepifanio) [reply components/filters/stories/Default/Default.vue]: Hey @denisonishchenkou ,

So the results of the filter will show only max of the results/data
in this case, if the same card has different locations in the card would have an array for the locations.
EG:

Locations selected:
Sydney / cbd
Sydney / Bondi

Card results:
1 card
Faculty of History
Location City: Sydney
Campus: CBD / Bondi
Location City: Melbourne
Campus: Southbank, Parkville

Something like that

[the remark this answers is not in the document]
Comment (denisonishchenkou) [reply components/filters/stories/Default/Default.vue]: > Options prop: what needs to be in this option to work? could it be anything? how do I tell the filter in this array of options I am sending to render and filter? Also, add one example of the options array otherwise who looks at the docs wouldn't understand how to use the component.
And if the need to refer how to use options going to the FilterDropdown component please add it more clearly so it's easy to understand. 

I've updated the link to the FilterDropdown component (with options and placeholderLabel examples) in order not to duplicate those examples.

> Inside of the options array in the label, the number between ( ) it shouldn't be in the data, should be calculated, so in the documentation also should explain how that works, and how that number will appear in there.

This doesn't mean that the data object is static, it's just an API of the component and I thought that there is no need to move this value to the additional field, labels can be created in the parent component, or even fetched from the BE side (you still have to fetch this data and maybe it's better to fetch the entire options object rather then fetch only quantities and combine them with labels from static object, stored in the FE app) I don't think it's required to divide the label string because there is no any differences in the design of the main label text and quantity and there is no any additional logic inside this component related to the quantity label. I can specify how I merged quantity with labels in the Default usage example, but I don't think this information sohuld be added to the FilterDropdown component readme. Does this make sense? Do you think it's better to add additional property to the option object (maybe "addon" or something similar) and render it in round brackets?

[the remark this answers is not in the document]
Comment (gepifanio) [reply components/filters/stories/Default/Default.vue]: Yes, I understand what you mean and makes sense, what I was saying is that this label with a count of results values shouldn't be static / hard coded, and the filters should calculate it automatically, that's why I suggested that but know I think I get the idea that's just the simulation of the API sending the data.

**Comment by denisonishchenkou** [comment]
@gepifanio I've finished requested changes. Could you please review?

File: components/error-box/ErrorBox.vue [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    <div
+      v-for="(message, messageIndex) in messages"
+      :key="messageIndex">
+      {{ message }}
+    </div>
+  </div>
+</template>
+
+<script>
+// TODO add styling
```
Comment (gepifanio) [inline components/error-box/ErrorBox.vue:14]: Could you delete this comment if the style has been added

Comment (gepifanio) [inline components/error-box/ErrorBox.vue:14]: Could you delete this comment if the style has been added

File: components/filters/components/filter-box/stories/FilterBoxWithMultipleFilters.vue [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+      this.filters[name] = changedOptions;
+    },
+    onClear(name) {
+      if (name) {
+        this.filters[name] = cloneDeep(defaultFilters[name]);
+      } else {
+        this.filters = cloneDeep(defaultFilters);
+      }
+    },
+    onUpdate() {
```
Comment (gepifanio) [inline components/filters/components/filter-box/stories/FilterBoxWithMultipleFilters.vue:72]: It is not being used, you could delete it.

File: components/filters/stories/MajorsAndSpecialisations/mockResults.js [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+
+      groupedByStudyOptionByExcerpt[studyOption] = groupBy(resultsArray, 'excerpt');
+
+      return groupedByStudyOptionByExcerpt;
+    },
+    {}
+  ),
+};
+
+// TODO calculate recursively
```
Comment (gepifanio) [inline components/filters/stories/MajorsAndSpecialisations/mockResults.js:258]: remove the comment TODO if it has been already done.

File: components/error-box/ErrorBox.vue (hunk 12 above)
Comment (denisonishchenkou) [reply components/error-box/ErrorBox.vue]: Done

File: components/filters/components/filter-box/stories/FilterBoxWithMultipleFilters.vue (hunk 13 above)
Comment (denisonishchenkou) [reply components/filters/components/filter-box/stories/FilterBoxWithMultipleFilters.vue]: Done

File: components/filters/stories/MajorsAndSpecialisations/mockResults.js (hunk 14 above)
Comment (denisonishchenkou) [reply components/filters/stories/MajorsAndSpecialisations/mockResults.js]: @gepifanio done

**Comment by gepifanio** [comment]
@jonathan-doc comments:

- [ ] The result counts aren't showing for Locations, and the Faculties result count (which does show) is dependent on the Locations

- [ ] The result counts should display the total number of results, independent of other filters

**Comment by denisonishchenkou** [comment]
@gepifanio I've finished the FilterWidget component, could you please review?
